extensions/amp-audio/1.0/test/test-amp-audio.js

```diff
@@ -0,0 +1,388 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpAudio} from '../amp-audio';
+// import {htmlFor} from '#core/dom/static-template';
+import {toggleExperiment} from '#experiments';
+import {waitFor} from '#testing/test-helper';
+import {naturalDimensions_} from '../../../../src/static-layout';
+
+describes.realWin(
+  'amp-audio-v1.0',
+  {
+    amp: {
+      extensions: ['amp-audio:1.0'],
+    },
+  },
+  (env) => {
+    let win;
+    let doc;
+    // let html;
+    let element;
+    let ampAudio;
+
+    beforeEach(async () => {
+      win = env.win;
+      doc = win.document;
+      // html = htmlFor(doc);
+      toggleExperiment(win, 'bento-audio', true, true);
+    });
+
+    /**
+     * Wait for iframe to be mounted
+     */
+    const waitForRender = async () => {
+      await element.buildInternal();
+      const loadPromise = element.layoutCallback();
+      const {shadowRoot} = element;
+      await waitFor(
+        () => shadowRoot.querySelector('audio'),
+        'audio tag mounted'
+      );
+      await loadPromise;
+    };
+
+    function getAmpAudio(attributes, opt_childNodesAttrs) {
+      ampAudio = doc.createElement('amp-audio');
+      for (const key in attributes) {
+        ampAudio.setAttribute(key, attributes[key]);
+      }
+      if (opt_childNodesAttrs) {
+        opt_childNodesAttrs.forEach((childNodeAttrs) => {
+          let child;
+          if (childNodeAttrs.tag === 'text') {
+            child = doc.createElement('p');
+          } else {
+            child = doc.createElement(childNodeAttrs.tag);
+            for (const key in childNodeAttrs) {
+              if (key !== 'tag') {
+                child.setAttribute(key, childNodeAttrs[key]);
+              }
+            }
+          }
+          ampAudio.appendChild(child);
+        });
+      }
+      doc.body.appendChild(ampAudio);
+      return ampAudio;
```

Comment:
It looks like this helper was taken from the 0.1 test suite, but here's some suggestions to clean it up a bit:
```suggestion
      ampAudio = createElementWithAttributes(doc, 'amp-audio', attributes);
      if (opt_childNodesAttrs) {
        opt_childNodesAttrs.forEach((childNodeAttrs) => {
          let child;
          if (childNodeAttrs.tag === 'text') {
            child = doc.createElement('p');
          } else {
            child = createElementWithAttributes(doc, childNodeAttrs.tag, childNodeAttrs);
          }
          ampAudio.appendChild(child);
        });
      }
      doc.body.appendChild(ampAudio);
      return ampAudio;
```